requirements.txt

```diff
@@ -1,4 +1,4 @@
-numpy>=1.24.4
+numpy~=1.24.4
```

Comment:
O_o, why not less than 2?

Reply:
For some reason, CI would still fail with [066218e](https://github.com/fairlearn/fairlearn/pull/1384/commits/066218e7c9fe9bf1138db6f450d8c01277a7517f). This is how I realised we have `numpy` API incompatibilities in the tests. Changing it to [937e09c](https://github.com/fairlearn/fairlearn/pull/1384/commits/937e09c6ff9f39a48727d0afe70a624ae7d5af77) makes CI pass. I was also confused why this is the case, but as I am already working on figuring this out on a more deeper level I though it won't hurt temporarily :)

Reply:
I reverted to range [4cbaf64](https://github.com/fairlearn/fairlearn/pull/1384/commits/4cbaf6458c873a9efc302a1a74132d9a8bc9a53e), which should fail only the `tensorflow` `build-test/other-ml` build. That one only seems to pass if we keep the version low as here `numpy~=1.24.4`.